signer/src/storage/postgres/read.rs

```diff
@@ -2492,879 +2509,850 @@ impl super::DbRead for PgStore {
-            , prevout_output_index  AS (SELECT ROW_NUMBER() OVER (), prevout_output_index FROM UNNEST($4::INTEGER[]) AS prevout_output_index)
-            , sighash               AS (SELECT ROW_NUMBER() OVER (), sighash FROM UNNEST($5::BYTEA[]) AS sighash)
-            , prevout_type          AS (SELECT ROW_NUMBER() OVER (), prevout_type FROM UNNEST($6::sbtc_signer.prevout_type[]) AS prevout_type)
-            , validation_result     AS (SELECT ROW_NUMBER() OVER (), validation_result FROM UNNEST($7::TEXT[]) AS validation_result)
-            , is_valid_tx           AS (SELECT ROW_NUMBER() OVER (), is_valid_tx FROM UNNEST($8::BOOLEAN[]) AS is_valid_tx)
-            , will_sign             AS (SELECT ROW_NUMBER() OVER (), will_sign FROM UNNEST($9::BOOLEAN[]) AS will_sign)
-            , x_only_public_key     AS (SELECT ROW_NUMBER() OVER (), x_only_public_key FROM UNNEST($10::BYTEA[]) AS x_only_public_key)
-            INSERT INTO sbtc_signer.bitcoin_tx_sighashes (
-                  txid
-                , chain_tip
-                , prevout_txid
-                , prevout_output_index
-                , sighash
-                , prevout_type
-                , validation_result
-                , is_valid_tx
-                , will_sign
-                , x_only_public_key
-            )
-            SELECT
-                txid
-              , chain_tip
-              , prevout_txid
-              , prevout_output_index
-              , sighash
-              , prevout_type
-              , validation_result
-              , is_valid_tx
-              , will_sign
-              , x_only_public_key
-            FROM tx_ids
-            JOIN chain_tip USING (row_number)
-            JOIN prevout_txid USING (row_number)
-            JOIN prevout_output_index USING (row_number)
-            JOIN sighash USING (row_number)
-            JOIN prevout_type USING (row_number)
-            JOIN validation_result USING (row_number)
-            JOIN is_valid_tx USING (row_number)
-            JOIN will_sign USING (row_number)
-            JOIN x_only_public_key USING (row_number)
-            ON CONFLICT DO NOTHING"#,
+    async fn get_bitcoin_canonical_chain_tip_ref(
+        &self,
+    ) -> Result<Option<model::BitcoinBlockRef>, Error> {
+        let mut tx = self.tx.lock().await;
+        PgRead::get_bitcoin_canonical_chain_tip_ref(tx.as_mut()).await
+    }
+
+    async fn get_stacks_chain_tip(
+        &self,
+        bitcoin_chain_tip: &model::BitcoinBlockHash,
+    ) -> Result<Option<model::StacksBlock>, Error> {
+        let mut tx = self.tx.lock().await;
+        PgRead::get_stacks_chain_tip(tx.as_mut(), bitcoin_chain_tip).await
+    }
+
+    async fn get_pending_deposit_requests(
+        &self,
+        chain_tip: &model::BitcoinBlockHash,
+        context_window: u16,
+        signer_public_key: &crate::keys::PublicKey,
```

Comment:
```suggestion
        signer_public_key: &PublicKey,
```
Same thing elsewhere too